numpy/linalg/linalg.py

```diff
@@ -2112,6 +2112,9 @@ def norm(x, ord=None, axis=None, keepdims=False):
     """
     x = asarray(x)
 
+    if not issubclass(x.dtype.type, floating):
```

Comment:
Ah, you want to check `complexfloating` too. Sure, sorry wasn't thinking about `complex`.